cmd/operator-sdk/generate/csv.go

```diff
@@ -74,14 +69,28 @@ Configure CSV generation by writing a config file 'deploy/olm-catalog/csv-config
 		},
 	}
 
-	cmd.Flags().StringVar(&c.csvVersion, "csv-version", "", "Semantic version of the CSV")
+	cmd.Flags().StringVar(&c.csvVersion, "csv-version", "",
+		"Semantic version of the CSV")
 	if err := cmd.MarkFlagRequired("csv-version"); err != nil {
 		log.Fatalf("Failed to mark `csv-version` flag for `generate csv` subcommand as required: %v", err)
 	}
 	cmd.Flags().StringVar(&c.fromVersion, "from-version", "",
 		"Semantic version of an existing CSV to use as a base")
-	cmd.Flags().StringVar(&c.csvConfigPath, "csv-config", "",
-		"Path to CSV config file. Defaults to deploy/olm-catalog/csv-config.yaml")
+	cmd.Flags().StringToStringVar(&c.inputs, "inputs",
+		map[string]string{
+			gencatalog.DeployDirKey: "deploy",
+			gencatalog.APIsDirKey:   filepath.Join("pkg", "apis"),
+		},
```

Comment:
WDYT about have 2 flags instead of combining all in the same one?
Would not be more simple to do the impl and to use/understanding as well have 2 flags one for each input instead of combining them?

Reply:
@estroz and I've discussed this before.
We want to keep the number of flags to a minimum otherwise it's going to add up and essentially become the CSV config file all over again.
That's why we're refactoring this to be a generator format with Inputs and Output.
There might be more inputs later on for CSV generation e.g CRD manifest locations.

Reply:
IHMO: This approach makes the user usability and implementation very complex and it will be harder if the quantity of the options grows as you said. 

- What is the advantage/facility for the user to write/read/use `--inputs [key]=value,[key]=value` instead of `--flag=value --flag=value`?
- Also, how we will gen the doc for all possible options since it is done dynamically? 
- Note that a user expected have the full info over the command by running `--help`. How can we describe each key option and the expected value following this approach?
- If we decide to deprecate an option, how it will works? Note that users will expected check that an option is deprecated by using the helper as follows which is achieved by using  `cmd.Flags().MarkDeprecated` such as [here](https://github.com/kubernetes/kubernetes/pull/73787/files#diff-7650a51eb24776a0115b675dd3842d63R177-R178). 

> Flag --config has been deprecated, Deprecated flag config

Sorry, but I still think that is not a good idea we follow with this approach.

Reply:
@camilamacedo86 That's a fair point actually. I think when I took over this PR, the motivation for a single `--inputs` or `--inputPaths` flag was so we could closely follow the generator pattern that controller-tools generators have.

But since then @estroz and I have agreed that the generator pattern is probably not necessary for our use case.
See: https://github.com/operator-framework/operator-sdk/pull/2511#discussion_r378385371

Since we're going to refactor the Generator config later anyway, I think removing the key/value `--inputs` flag in favor of cmd specific flags like `api-dir`, `deploy-dir` would actually be better.
So I'll make that change in this PR.

Reply:
It is very cool 👍!! really tks for share the info as well.
You are terrific @hasbro17 @estroz 
Fell free to close it as resolve.